src/repository/models.py

```diff
@@ -0,0 +1,1033 @@
+STAGE_PREPRINT_UNSUBMITTED = 'preprint_unsubmitted'
+STAGE_PREPRINT_REVIEW = 'preprint_review'
+STAGE_PREPRINT_PUBLISHED = 'preprint_published'
+STAGE_PREPRINT_REJECTED = 'preprint_rejected'
+
+SUBMITTED_STAGES = {
+    STAGE_PREPRINT_REVIEW,
+    STAGE_PREPRINT_PUBLISHED,
+    STAGE_PREPRINT_REJECTED
+}
+
+
+def html_input_types():
+    return (
+        ('text', 'Text'),
+        ('select', 'Dropdown'),
+        ('checkbox', 'Checkbox'),
+        ('number', 'Number'),
+        ('date', 'Date'),
+        ('textarea', 'Text Area'),
+    )
+
+
+def width_choices():
+    return (
+        (3, '3'),
+        (6, '6'),
+        (9, '9'),
+        (12, '12'),
+    )
+
+
+fs_path = os.path.join('files/')
+preprint_file_store = JanewayFileSystemStorage(location=fs_path)
+preprint_media_store = JanewayFileSystemStorage()
+
+
+def preprint_file_upload(instance, filename):
+    try:
+        uuid_filename = str(uuid.uuid4()) + '.' + str(filename.split('.')[1])
+    except IndexError:
+        uuid_filename = str(uuid.uuid4())
+
+    path = os.path.join('repos', str(instance.preprint.pk), uuid_filename)
+    instance.original_filename = filename
+    return path
+
+
+def repo_media_upload(instance, filename):
+    try:
+        filename = str(uuid.uuid4()) + '.' + str(filename.split('.')[1])
+    except IndexError:
+        filename = str(uuid.uuid4())
+
+    path = "repos/{0}/".format(instance.pk)
+    return os.path.join(path, filename)
+
+
+class Repository(model_utils.AbstractSiteModel):
+    press = models.ForeignKey('press.Press')
+    name = models.CharField(max_length=255)
+    short_name = models.CharField(
+        max_length=15,
+        help_text='Shortened version of the name eg. olh. Max 15 chars.'
+    )
+    object_name = models.CharField(
+        max_length=255,
+        help_text='eg. preprint or article',
+    )
+    object_name_plural = models.CharField(
+        max_length=255,
+        help_text='eg. preprints or articles',
+    )
+    managers = models.ManyToManyField('core.Account', blank=True)
+    logo = models.ImageField(
+        blank=True,
+        null=True,
+        storage=preprint_media_store,
+        upload_to=repo_media_upload,
+    )
+    favicon = models.ImageField(
+        blank=True,
+        null=True,
+        storage=preprint_media_store,
+        upload_to=repo_media_upload,
+    )
+    hero_background = models.ImageField(
+        blank=True,
+        null=True,
+        storage=preprint_media_store,
+        upload_to=repo_media_upload,
+    )
+    publisher = models.CharField(
+        max_length=255,
+        help_text=_('Used for outputs including DC and Citation metadata'),
+    )
+    custom_js_code = models.TextField(
+        blank=True,
+        null=True,
+        help_text=_('The contents of this field are output into the JS area'
+                    'at the foot of every Repository page.')
+    )
+    live = models.BooleanField(
+        default=False,
+        verbose_name='Repository is Live?'
```

Comment:
Might need help text here saying that the repository won't be accessible by anyone otherwise (not even managers)